### DISCUSSION TITLE
Best method for future transactions ?

### DISCUSSION THREAD

**Discussion by M-Paladin** [firefly-iii/firefly-iii#4552] [General]
I am running Firefly III version 5.4.6

**Description**

I would like to know which is the best method to have a good vision for future transaction over the period.
I know that I have many transactions that are nearly the same from month to month (date and amount may change).
I want to automate that.
For example :
- 27th : salary (amount vary a little)
- 29th : Internet subscription debit (same amount)
- 30th : Deferred Credit Card debit (amount vary from single to double)
- 1st : transfer to another bank (same amount, vary sometimes)
- 3rd : transfer from another bank (same amount)
- 5th : rent paid by tenant (same amount)
- 14th : loan (same amount)
- 15th : property tax (same amount)
- ...

So around the 15-16th, I would like to have the view for the next period from the 27th of the current month to the 26th of the next month.
Once this done I know if I have to transfer money from my saving to my account to be in positive at the end of the period or on the contrary if I can transfer money to my saving as soon as possible

I thought recurring transactions would be better but I'm not really sure because crontab and "force" option may create duplicate if I use them many times. Furthermore, once executed, the date of the next occurrence does not change. It stays the same until the date is passed.
I also check bills section but it doesn't seem to fit all my needs.
There are also the rules which seems interesting to combine with.
I think I don't need budget because I don't want to save money each month. I just want my account to be negative at no time.

What do you think I should do ?
I began using Firefly III instead of MoneyManagerEx because it seemed very interesting and promising with good evolutions and a pleasant interface but now I not really sure I'm using it right and if it fits all my needs.
What do you think about this ?

Thank you in advance.

**Extra info**
 

**Bonus points**

- I have read the FAQ at https://bit.ly/FF3-FAQ

**Comment by 4d1830** [comment]
Dear all, 

me too I'm interested to this topic.

My goal is to have a complete view of future bills in the next period, eg. six months or twelve months, so I can have an idea of which impact can have bills on my expenses.

How I can achieve this goal?

Thank's a lot!

**Reply by JC5** [reply]
https://docs.firefly-iii.org/firefly-iii/about-firefly-iii/what-its-not/#prediction-forecasting-future-gazing

**Comment by M-Paladin** [comment]
For the moment, I'm testing a way to do it, at least the way I want to.
First I create my recurring transactions in Firefly but I don't generate the corresponding transactions with cron. It's only used as "model"
So I created my own bash script which uses API.
First I get the complete list of recurring transactions.
Next I test each one to see if the first date is in the period I want to generate.
After that I create a transaction based on the recurring transaction.
Then I test if the recurring transaction should still exist or not (last date or last repetition)
If it should, I update the recurring transaction, change the first date with the next occurring date, reduce the number of repetitions if needed.
If it shouldn't, I delete the recurring transaction.

**Comment by Hohenbostler** [comment]
Hello i´m really new in Firefly and this Discussions is just what i search.
I filled Standing Orders for every Period ( like every month ) an so the System could know what are my future transanctions should be.

Is there know any way to see this future Transaction in a overview ?

**Comment by brunofontes** [comment]
I am adopting Firefly, and I'm having a similar issue. Most of the time, I'll schedule some bills to be paid in the correct date (for multiple reasons).

But this is causing me an issue: my net worth will not consider that scheduled translation by the period, but it does not appear in my Bills section as well, as it "was already paid". So it appears I have more money that I actually have.

The best way to solve this, is going into the Bills section, summing up the future bills that are already paid, go back to the main screen and subtract that from my Net Worth. It works, but is still hard to do when you just want to know how much money you still have available for the month.

Does anyone have a better idea on how to handle this?

Thanks in advance!
